samples/bluetooth/fast_pair/locator_tag/src/factory_reset.c

```diff
@@ -88,6 +87,24 @@ int bt_fast_pair_factory_reset_user_action_perform(void)
 	return 0;
 }
 
+static void factory_reset_prepare(void)
+{
+	STRUCT_SECTION_FOREACH(app_factory_reset_callbacks, cbs) {
+		if (cbs->prepare) {
+			cbs->prepare();
+		}
+	}
+}
+
+static void factory_reset_executed(void)
+{
+	STRUCT_SECTION_FOREACH(app_factory_reset_callbacks, cbs) {
+		if (cbs->executed) {
```

Comment:
Maybe we could pass information about error here as callback parameter. Thanks to that listeners would always be aware of factory reset state. Also, main app module could control the `k_panic` call on factory reset failure.

Reply:
Having multiple callback may also allow us to untangle factory reset dependency on `fp_adv` and `fast_pair` subsystem - it would be reasonable to move those API calls to callback now I think

Reply:
`app_fp_adv_disable` may not be needed as part of `fp_adv` API too

Reply:
`app_fp_adv_is_ready` could also be removed from the `fp_adv` API

Reply:
Discussed, will create a follow-up based on this suggestions, for now we will keep the implementation as is.

Reply:
Please link the follow-up ticket so I can resolve my comment here.

Reply:
https://nordicsemi.atlassian.net/browse/NCSDK-28552